infra/logging/src/main/java/com/simprints/infra/logging/Simber.kt

```diff
@@ -119,13 +138,51 @@ object Simber {
      * 1 kB in size."
      */
     fun tag(tag: String, isUserProperty: Boolean = false): Simber {
-        if (isUserProperty)
-            Timber.tag(USER_PROPERTY_TAG + tag)
-        else
-            Timber.tag(tag)
+        var conformingTag = tag
+
+        // Firebase Analytics requires tag to be no longer than 24 characters
+        var maxLength = 24
+        if (isUserProperty) {
+            conformingTag = USER_PROPERTY_TAG + conformingTag
+            // Firebase Analytics requires user property tag to be no longer than 40 characters
+            maxLength = 40
+        }
+        conformingTag = ensureMaxLength(conformingTag, maxLength)
+        conformingTag = ensureCharactersAreValid(conformingTag)
+
+        Timber.tag(conformingTag)
         return Simber
     }
 
+    /*
+    * Check that tag complies with Firebase Analytics requirements:
+    * Name must consist of letters, digits or _ (underscores).
+     */
+    private fun ensureCharactersAreValid(tag: String): String {
+        var validTag = tag
+        if (validTag.contains(Regex("[^a-zA-Z0-9_]"))) {
```

Comment:
You are compiling 2 regex instances per call. I would suggest extracting the regex out of the method entirely and reusing the same instance.

Reply:
Also, the variable could be avoided since you can call `contains()` on the tag param and return the result of `replace()` or the initial tag as-is directly.